packages/material-ui-styles/src/makeStyles/makeStyles.d.ts

```diff
@@ -71,5 +72,5 @@ export type StylesHook<S extends Styles<any, any>> = StylesRequireProps<S> exten
 
 export default function makeStyles<S extends Styles<any, any>>(
   styles: S,
-  options?: WithStylesOptions,
+  options?: WithStylesOptions<ThemeOfStyles<S>>,
```

Comment:
I remembered a similar issue from before. We have these helpers to force inference. I guess the type checker will stop inference for `T` for any `interface<T, U<T>>`.